spark/src/test/scala/org/apache/spark/sql/delta/CustomCatalogSuite.scala

```diff
@@ -297,6 +297,22 @@ class CustomCatalogSuite extends QueryTest with SharedSparkSession
       }
     }
   }
+
+  test("custom catalog that generates location for managed tables") {
+    // Reset catalog manager so that the new `spark_catalog` implementation can apply.
+    spark.sessionState.catalogManager.reset()
+    withSQLConf("spark.sql.catalog.spark_catalog" -> classOf[DummySessionCatalog].getName) {
+      withTable("t") {
+        withTempPath { path =>
+          sql(s"CREATE TABLE t (id LONG) USING delta TBLPROPERTIES (fakeLoc='$path')")
+          val t = spark.sessionState.catalogManager.v2SessionCatalog.asInstanceOf[TableCatalog]
+            .loadTable(Identifier.of(Array("default"), "t"))
+          // It should be a managed table.
+          assert(!t.properties().containsKey(TableCatalog.PROP_EXTERNAL))
```

Comment:
can we also test that if the catalog does not support managed tables, then the create table can throw error?

Reply:
I can't test this here as the underlying catalog is Spark HMS which we can't customize. I think the current test is OK as we have guaranteed the created table is managed table and if we replace HMS with UC, we can for sure fail correctly.